### PR TITLE
Adds remove path button - fixes #314

### LINKED ISSUE #314: Deleting paths
As an admin (Unleasher) or the owner of the profile I should be able to remove a given path.

Technically this should do a DELETE request to the given path API url (deleting not yet implemented on the API side).

### PR DESCRIPTION
**Pull request by jenriquejr**

### CODE REVIEW

**Comment by jenriquejr** [comment]
removing paths

File: app/actions/PathsActions.js [hunk 1]
```diff
@@ -78,6 +83,19 @@ export function pathsRename(pathId, newName) {
...
 }
 
+export function pathsRemove(pathId) {
+  return (dispatch) => {
+    dispatch({ type: PATHS.REMOVE.START, pathId });
+
+    return fetch(`${config.paths_api_url}/${pathId}`, {
+      method: 'delete',
+    })
+      .then((response) => (response.status >= 400 ? Promise.reject(response.text()) : null))
```
Comment (szymonmichalak) [inline app/actions/PathsActions.js]: Could you please explain what's going on here? :-)

File: app/reducers/pathsReducer.js [hunk 2]
```diff
@@ -51,6 +62,12 @@ function pathsReducer(state = initialState, action) {
       return { ...state, errors, loading: false, list: merge(state.list, [paths]) };
     case PATHS.UPDATE.FAILURE:
       return { ...state, errors, loading: false };
+    case PATHS.REMOVE.START:
+      return { ...state, errors, loading: true };
+    case PATHS.REMOVE.SUCCESS:
+      return { ...state, errors, loading: false, list: filterPathList(state.list, action.pathId) };
+    case PATHS.REMOVE.FAILURE:
+      return { ...state, errors, loading: false };
```
Comment (szymonmichalak) [inline app/reducers/pathsReducer.js]: While you're changing `pathsReducer`, could you please change `loading` to `isLoading` here? It's the only reducer that is inconsistent...

File: app/actions/PathsActions.js (hunk 1 above)
Comment (jenriquejr) [inline app/actions/PathsActions.js]: Yeah, I am not sure if we also need this in other dispatchers, this is for errors in server. If the server returns any error code the promise will not be rejected automatically.
This also assumes that if the server returned 200 the delete was successful.

Comment (szymonmichalak) [inline app/actions/PathsActions.js]: I see... we for sure need this handler, but I would suggest to make a wrapper for `fetch` and handle those errors there. But maybe let's do it in a separate PR - I can come up with some proposal :-)

Comment (jenriquejr) [inline app/actions/PathsActions.js]: sounds good to me - I can take care of it if on a separate PR - I don't oppose if you want to do it either.

Comment (szymonmichalak) [inline app/actions/PathsActions.js]: Great, I will prepare my proposal and let you know!

File: app/reducers/pathsReducer.js (hunk 2 above)
Comment (jenriquejr) [inline app/reducers/pathsReducer.js]: updated

**Comment by karolsojko** [comment]
@jenriquejr you have lint errors - check the Travis builds for details

File: app/actions/PathsActions.js (hunk 1 above)
Comment (szymonmichalak) [inline app/actions/PathsActions.js]: That's what I had in mind: https://github.com/x-team/unleash/pull/337 - @jenriquejr could you please take a look?

**Comment by jenriquejr** [comment]
@karolsojko I fixed the new one but the following one is a false warning
```sh
/var/www/unleash/app/components/Path.js
39:22  error  'loggedInUser' is missing in props validation  react/prop-types
```
see here:
https://github.com/x-team/unleash/pull/334/files#diff-a8b854e8822cd835fb740814e3f13a08R83

**Comment by karolsojko** [comment]
@jenriquejr haven't came across false warnings before - can you check what's causing it ?

**Comment by jenriquejr** [comment]
@karolsojko I can not replicate it locally - it's pretty weird can you clear travis cache and rerun ?

**Comment by jenriquejr** [comment]
screen shot 2017-01-23 at 12 51 55 pm

**Comment by jenriquejr** [comment]
@karolsojko reordered the `propTypes` properties. This time it did not throw any false warnings

**Pull request merged**